test/integration/plants/plant_test.go

```diff
@@ -0,0 +1,250 @@
+// Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package plants_test
+
+import (
+	"context"
+	"encoding/base64"
+	"flag"
+	"fmt"
+	"time"
+
+	"k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	. "github.com/gardener/gardener/test/integration/shoots"
+
+	gardencorev1alpha1 "github.com/gardener/gardener/pkg/apis/core/v1alpha1"
+	"github.com/gardener/gardener/pkg/apis/garden/v1beta1"
+	"github.com/gardener/gardener/pkg/logger"
+	. "github.com/gardener/gardener/test/integration/framework"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var (
+	kubeconfig        = flag.String("kubeconfig", "", "the path to the kubeconfig  of the garden cluster that will be used for integration tests")
+	shootName         = flag.String("shootName", "", "the name of the shoot we want to test")
+	shootNamespace    = flag.String("shootNamespace", "", "the namespace name that the shoot resides in")
+	testShootsPrefix  = flag.String("prefix", "", "prefix to use for test shoots")
+	logLevel          = flag.String("verbose", "", "verbosity level, when set, logging level will be DEBUG")
+	shootTestYamlPath = flag.String("shootpath", "", "the path to the shoot yaml that will be used for testing")
```

Comment:
Why do we need this parameter? I thought we are testing `Plant`s here, not `Shoot`s?

Reply:
We need to first create a shoot to use as a Plant, thats why we need it here.

Reply:
Why? We don't want to couple `Shoot`s to `Plant`s, there is no need for that, is it?

Reply:
Which external cluster to use as A Plant would you suggest otherwise?

Reply:
We are not really coupling, we just need a Kubernetes cluster that we can add as a plant and we thought of using shoots as plants for testing. We could as well use existing clusters but we want the test to be able to create clusters as well for our testing purposes.

Reply:
> We are not really coupling

Yes, you are

> Which external cluster to use as A Plant would you suggest otherwise?

> we just need a Kubernetes cluster that we can add as a plant and we thought of using shoots as our mean to get clusters to register as plants for testing.

Any other, it shouldn't matter to the test. It can be a shoot, why not, that's totally fine and makes sense, but the test should not depend on it.

Reply:
@danielfoehrKn we can get use the test definitions as recommended by @rfranzke or and just supply a kubeconfig without the need to create the shoot in the test.